ethpm/contract.py

```diff
@@ -31,45 +30,35 @@ def __init__(self, address: bytes = None, **kwargs: Dict[str, Any]) -> None:
 
     @classmethod
     def constructor(self, *args: Any, **kwargs: Any) -> bool:
-        if self.has_linkable_bytecode() and not self.is_bytecode_linked:
+        if self.needs_bytecode_linking():
             raise BytecodeLinkingError(
                 "Contract cannot be deployed until its bytecode is linked."
             )
         return super(LinkableContract, self).constructor(*args, **kwargs)
 
-    @combomethod
-    def has_linkable_bytecode(self) -> bool:
-        """
-        Return a boolean indicating the presence of linkable bytecode
-        on this contract factory or instance.
-        """
-        if self.deployment_link_refs or self.runtime_link_refs:
-            return True
-        return False
-
-    @combomethod
-    def linked_contract_types(self) -> List[str]:
+    @classmethod
```

Comment:
Ah, I see now that this has to be a class method.  How about computing this at the time of class creation?  This *could* be done either within the `Contract.factory` method or as a metaclass (but probably easier if done in the `Contract.factory`)